extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,272 @@
+            Type _ when targetType == typeof(PageBlobClient[]) => blobCollection.Select(b => (PageBlobClient)b).ToArray(),
+            Type _ when targetType == typeof(AppendBlobClient[]) => blobCollection.Select(b => (AppendBlobClient)b).ToArray(),
+            Type _ when targetType == typeof(BlobContainerClient) => blobCollection.Select(b => (BlobClient)b).ToArray(),
+            _ => null
+        };
+
+        private IEnumerable<object>? ToTargetTypeCollection(Type targetType, IEnumerable<object> blobCollection) => targetType switch
+        {
+            Type _ when targetType == typeof(IEnumerable<String>) || targetType == typeof(String[]) => blobCollection.Select(b => (string)b),
+            Type _ when targetType == typeof(IEnumerable<Stream>) => blobCollection.Select(b => (Stream)b),
+            Type _ when targetType == typeof(IEnumerable<Byte[]>) => blobCollection.Select(b => (Byte[])b),
+            Type _ when targetType == typeof(IEnumerable<BlobBaseClient>) => blobCollection.Select(b => (BlobBaseClient)b),
+            Type _ when targetType == typeof(IEnumerable<BlobClient>) => blobCollection.Select(b => (BlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<BlockBlobClient>) => blobCollection.Select(b => (BlockBlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<PageBlobClient>) => blobCollection.Select(b => (PageBlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<AppendBlobClient>) => blobCollection.Select(b => (AppendBlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<BlobContainerClient>) => blobCollection.Select(b => (BlobClient)b),
+            _ => null
+        };
+
+        private async Task<object?> DeserializeToTargetObjectAsync(Type targetType, string connectionName, string containerName, string blobName)
+        {
+            var content = await GetBlobStreamAsync(connectionName, containerName, blobName);
+            return _workerOptions.Value.Serializer.Deserialize(content, targetType, CancellationToken.None);
+        }
+
+        private object? DeserializeToTargetObjectCollection(IEnumerable<object> blobCollection, Type targetType)
+        {
+            (string methodName, Type type) = targetType.IsArray
+                                            ? (nameof(CloneToArray), targetType.GetElementType())
+                                            : (nameof(CloneToList), targetType.GenericTypeArguments[0]);
+
+            blobCollection = blobCollection.Select(b => Convert.ChangeType(b, type));
+            MethodInfo method = typeof(BlobStorageConverter).GetMethod(methodName);
+            MethodInfo genericMethod = method.MakeGenericMethod(type);
+
+            return genericMethod.Invoke(null, new[] { blobCollection.ToList() });
+        }
+
+        public static T[] CloneToArray<T>(IList<object> source)
```

Comment:
Updated both methods (`CloneToArray`, `CloneToList`) to public from private for keeping E2E working. We can discuss better approach here.

Reply:
Why did you need to update them? Was the `GetMethod` reflection call not finding them? If so, you need to pass in some flags to `GetMethod` to tell it to search for `private static`.